flatliners/prometheusendpoint.py

```diff
@@ -0,0 +1,51 @@
+from time import sleep, time
+from collections import defaultdict
+from .baseflatliner import BaseFlatliner
+
+from prometheus_client import Gauge, start_http_server
+
+# Set up logging
+import logging
+_LOGGER = logging.getLogger(__name__)
+
+class PrometheusEndpoint(BaseFlatliner):
+    def __init__(self):
+        super().__init__()
+        self.pruning_interval = 5*60 # seconds # This value should be close to the scraping interval for your prometheus
+        _LOGGER.info("Prometheus Endpoint initialized. Metric pruning interval is {0} seconds".format(self.pruning_interval))
+        self.published_metric_timestamps = defaultdict(list)
+        self.weirdness_score_gauge = Gauge('weirdness_score','Weirdness score for the given Cluster and Version',['cluster','version'])
+
+    def on_next(self, x):
+        try:
+            self.weirdness_score_gauge.labels(cluster=str(x.cluster), version=str(x.version)).set(x.weirdness_score)
+            self.published_metric_timestamps[str(x.cluster)] = [int(time()),str(x.version)] # timestamp when the metric was published and metric version info
+        except Exception as e:
+            _LOGGER.error("Couldn't process the following packet {0}. Reason: {1}".format(x,str(e)))
+            raise e
+        pass
+
+    def _delete_prev_metrics(self):
```

Comment:
for clarity and naming. `_delete_stale_metrics` is better. `prev` is too generic